src/kakarot/execution_context.cairo

```diff
@@ -364,7 +375,7 @@ namespace ExecutionContext {
             // this is called after a top level check that a given context is stopped
             // so this is the case of a stopped, non reverted context
             // meaning events should be fired off!
-            emit_events(self.events_len, self.events);
+            emit_events(self.evm_contract_address, self.events_len, self.events);
```

Comment:
this makes me wonder abt smth, let me give an example:
context A has subcontext B,

context A starts...,  then subcontext B is entered,

when context B terminates successfully at time t,

then context A resumes, and then reverts.

are subcontext B events emitted at time t?

or never emitted, since the overall call reverted?

Reply:
i think in this form they would still be emitted. 

can add 
- a check where emit_event doesnt get called unless its the finalized root calling context
   - if root, go through events and fire them
   - if not, push the event w/ the current execution context to the events of the calling stack

Reply:
I think it'll help, right?